### PR TITLE
Updated default collector version to 0.50.0

### PR DESCRIPTION
**Pull request by TylerHelmuth**
This PR updates the default collector image version to 0.50.0. Reviewing opentelemetry-collector and opentelemetry-collector-contrib I see no breaking changes that will affect the chart configuration.

Fixes https://github.com/open-telemetry/opentelemetry-helm-charts/issues/194

### CODE REVIEW

File: charts/opentelemetry-collector/examples/daemonset-collector-logs/values.yaml [hunk 1]
```diff
@@ -7,3 +7,10 @@ config:
   receivers:
     filelog:
       exclude: []
+  service:
+    pipelines:
+      logs:
+        receivers:
+          - otlp
+          - filelog
```
Comment (dmitryax) [inline charts/opentelemetry-collector/examples/daemonset-collector-logs/values.yaml:15]: Is this a related change?

Comment (dmitryax) [reply charts/opentelemetry-collector/examples/daemonset-collector-logs/values.yaml:15]: Do you mind moving it to a separate PR for clean git history?

Comment (TylerHelmuth) [reply charts/opentelemetry-collector/examples/daemonset-collector-logs/values.yaml:15]: Sure

Comment (TylerHelmuth) [reply charts/opentelemetry-collector/examples/daemonset-collector-logs/values.yaml]: reverted

**Review by dmitryax — APPROVED** [review]
